assets/js/base/components/block-error-boundary/index.js

```diff
@@ -14,7 +14,24 @@ class BlockErrorBoundary extends Component {
 	state = { hasError: false };
 
 	static getDerivedStateFromError( error ) {
-		return { errorMessage: error.message, hasError: true };
+		if (
+			typeof error.statusText !== 'undefined' &&
+			typeof error.status !== 'undefined'
+		) {
+			return {
+				errorMessage: (
+					<Fragment>
+						<strong>{ error.status }</strong>:&nbsp;
+						{ error.statusText }
+					</Fragment>
+				),
+				hasError: true,
+			};
+		}
+
+		if ( typeof error.message !== 'undefined' ) {
+			return { errorMessage: error.message, hasError: true };
+		}
 	}
```

Comment:
This returns undefined if none of the conditions match, should probably default to a more generic message as a fallback?

Reply:
The message is not required; we can return an undefined errorMessage though.